src/core/public/ui_settings/ui_settings_client.ts

```diff
@@ -124,12 +129,20 @@ You can use \`IUiSettingsClient.get("${key}", defaultValue)\`, which will just r
     );
   }
 
-  async set(key: string, value: any) {
-    return await this.update(key, value);
+  async getUserProvided<T = any>(key: string, scope: UiSettingScope) {
+    return await this.api.getWithScope(scope).then((response: any) => {
```

Comment:
Better to add a validation here to ensure the setting supports the scope passed in.